extension/src/popup/components/InternalTransaction/ReviewTransaction/index.tsx

```diff
@@ -96,6 +142,37 @@ export const ReviewTx = ({
     ? truncatedFedAddress(federationAddress)
     : truncatedPublicKey(destination);
 
+  // Scan transaction
+  const txScanResult = simulationState.data?.scanResult;
+  const shouldTreatTxAsUnableToScan = useShouldTreatTxAsUnableToScan();
+
+  // Compute warning state once - used by both label and expanded components
+  const isUnableToScan = shouldTreatTxAsUnableToScan(txScanResult);
+  const hasScanResult = !!txScanResult;
+
+  // Determine malicious/suspicious from scan result (not dev-only)
+  const isMalicious =
+    txScanResult?.validation &&
+    "result_type" in txScanResult.validation &&
+    txScanResult.validation.result_type === "Malicious";
+  const isSuspicious =
+    txScanResult?.validation &&
+    "result_type" in txScanResult.validation &&
+    txScanResult.validation.result_type === "Warning";
```

Comment:
It looks like `isMalicious` and `isSuspicious` grab the designation from `validation.result_type`. But `getTransactionWarningType` does the same thing. They seem duplicative of each other.